### PR TITLE
Misuv 5613 v2

### PR DESCRIPTION
**Pull request by avelx**

### CODE REVIEW

**Comment by simondrugan16** [comment]
need to remove printlns

File: app/connectors/FeedbackConnector.scala [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  private def partialsReadyHeaderCarrier(implicit request: Request[_]): HeaderCarrier = {
+    val hc = itvcHeaderCarrierForPartialsConverter.headerCarrierEncryptingSessionCookieFromRequest(request)
+    itvcHeaderCarrierForPartialsConverter.headerCarrierForPartialsToHeaderCarrier(hc)
+  }
+
+  def submit(formData: FeedbackForm)
+            (implicit request: Request[_]): Future[Either[Int, Unit]] = {
+    val ref: String = request.headers.get(REFERER).getOrElse("N/A")
+
+    println(s"DATA: ${feedbackServiceSubmitUrl}")
```
Comment (simondrugan16) [inline app/connectors/FeedbackConnector.scala:56]: remove println

File: app/controllers/feedback/FeedbackController.scala [hunk 2]
```diff
@@ -81,27 +80,23 @@ class FeedbackController @Inject()(implicit val config: FrontendAppConfig,
 
   def submit: Action[AnyContent] = auth.authenticatedAction(isAgent = false) {
     implicit request =>
+      println("AAAAAAA")
```
Comment (simondrugan16) [inline app/controllers/feedback/FeedbackController.scala:83]: remove println

File: test/controllers/FeedbackControllerSpec.scala [hunk 3]
```diff
@@ -119,13 +121,14 @@ class FeedbackControllerSpec extends MockAuthenticationPredicate
       "return an SEE_OTHER (303)" in {
         setupMockAgentAuthRetrievalSuccess(testAgentAuthRetrievalSuccess, withClientPredicate = false)
         mockBothIncomeSources()
-
+        println("AAAAAAA")
```
Comment (simondrugan16) [inline test/controllers/FeedbackControllerSpec.scala:124]: remove println

File: test/controllers/FeedbackControllerSpec.scala [hunk 4]
```diff
@@ -119,13 +121,14 @@ class FeedbackControllerSpec extends MockAuthenticationPredicate
...
         when(mockHttpClient.POSTForm[HttpResponse](any(), any(), any())(any(), any(), any())).thenReturn(Future.successful(HttpResponse(OK, "test")))
-
+        when(mockFeedbackConnector.submit(any())(any())).thenReturn(Future.successful(Right(())))
+        println("BBBBBBB")
         when(mockThankYouView(any(), any(), any())(any(), any(), any())).thenReturn(HtmlFormat.empty)
-
+        println("CCCCCC")
```
Comment (simondrugan16) [inline test/controllers/FeedbackControllerSpec.scala:129]: remove println

File: app/connectors/FeedbackConnector.scala [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+  def submit(formData: FeedbackForm)
+            (implicit request: Request[_]): Future[Either[Int, Unit]] = {
+    val ref: String = request.headers.get(REFERER).getOrElse("N/A")
+
+    http.POSTForm[HttpResponse](feedbackServiceSubmitUrl,
+      formData.toFormMap(ref))(readForm, partialsReadyHeaderCarrier, ec).map {
+      resp =>
+        resp.status match {
+          case OK =>
+            Logger("application").info(s"OK....")
```
Comment (avelx) [inline app/connectors/FeedbackConnector.scala:62]: if we can align messages in log info and error to the common standard we use across the service

File: app/controllers/feedback/FeedbackController.scala [hunk 6]
```diff
@@ -84,24 +83,19 @@ class FeedbackController @Inject()(implicit val config: FrontendAppConfig,
...
-                  itvcErrorHandler.showInternalServerError()
-              }
-          }
-        }.recover {
+        formData =>
+          feedbackConnector.submit(formData).flatMap {
+            case Right(_) =>
+              Future.successful(Redirect(routes.FeedbackController.thankYou))
+            case Left(status) =>
+              Logger("application").error(s"[Agent] Unexpected status code from feedback form: $status")
```
Comment (avelx) [inline app/controllers/feedback/FeedbackController.scala:91]: I guess this is not an Agent call so message in the log should reflect this

File: app/connectors/FeedbackConnector.scala [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+  def submit(formData: FeedbackForm)
+            (implicit request: Request[_]): Future[Either[Int, Unit]] = {
+    val ref: String = request.headers.get(REFERER).getOrElse("N/A")
+
+    http.POSTForm[HttpResponse](feedbackServiceSubmitUrl,
+      formData.toFormMap(ref))(readForm, partialsReadyHeaderCarrier, ec).map {
+      resp =>
+        resp.status match {
+          case OK =>
+            Logger("application").info(s"[IncomeTaxViewChangeConnector][submit] - RESPONSE status: ${resp.status}")
```
Comment (avelx) [inline app/connectors/FeedbackConnector.scala:62]: log message must be unique in the scope of the component

File: it/test/controllers/agent/FeedbackControllerISpec.scala [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+
+        stubAuthorisedAgentUser(authorised = true)
+
+        IncomeTaxViewChangeStub.stubGetIncomeSourceDetailsResponse(testMtditid)(200, noPropertyOrBusinessResponse)
+
+        When(s"I call GET /report-quarterly/income-and-expenses/view/agents/feedback")
+        val res: WSResponse = IncomeTaxViewChangeFrontend.getFeedbackPage(clientDetailsWithConfirmation)
+
+        res should have(
+          httpStatus(200),
```
Comment (avelx) [inline it/test/controllers/agent/FeedbackControllerISpec.scala:65]: if we can avoid magic number in the code base, there are predefined HttpStatus codes we can use

**Review by hnare — APPROVED** [review]

**Pull request merged**